Ds3/Calls/PutObjectRequest.cs

```diff
@@ -99,18 +100,32 @@ internal override Stream GetContentStream()
         public Guid JobId { get; private set; }
         public long Offset { get; private set; }
 
-        public PutObjectRequest(string bucketName, string objectName, Guid jobId, long offset, Stream content)
+        private void SetParameters(string bucketName, string objectName, Guid jobId, long offset)
         {
             this.BucketName = bucketName;
             this.ObjectName = objectName;
             this.JobId = jobId;
             this.Offset = offset;
-            this._content = content;
+
             if (jobId != Guid.Empty)
             {
                 QueryParams.Add("job", jobId.ToString());
                 QueryParams.Add("offset", offset.ToString());
             }
         }
+
+        public PutObjectRequest(string bucketName, string objectName, Guid jobId, long offset, Stream content, long length)
+        {
+            SetParameters(bucketName, objectName, jobId, offset);
+
+            this._content = new PutObjectRequestStream(content, Offset, length);
```

Comment:
I think I was imagining this as being passed in, rather than being a different constructor.